packages/web/src/pages/pay-and-earn-page/desktop/PayAndEarnPage.tsx

```diff
@@ -30,24 +36,31 @@ export const messages = {
   downloadCSV: 'Download CSV'
 }
 
-enum TableType {
-  SALES = 'sales',
-  PURCHASES = 'purchases',
-  WITHDRAWALS = 'withdrawals'
-}
-
 type TableMetadata = {
   label: string
   downloadCSV: () => void
   isDownloadCSVButtonDisabled: boolean
 }
 
-export const PayAndEarnPage = () => {
-  const [selectedTable, setSelectedTable] = useState<TableType>(TableType.SALES)
+export const PayAndEarnPage = ({ tableView }: PayAndEarnPageProps) => {
+  const dispatch = useDispatch()
   const { data: balance } = useUSDCBalance({
     isPolling: true,
     pollingInterval: 3000
   })
+  const accountHasTracks = useSelector(getAccountHasTracks)
+
+  const [tableOptions, setTableOptions] = useState<TableType[] | null>(null)
+  const [selectedTable, setSelectedTable] = useState<TableType | null>(null)
+  useEffect(() => {
+    if (accountHasTracks !== null) {
+      const tableOptions = accountHasTracks
+        ? [TableType.SALES, TableType.PURCHASES, TableType.WITHDRAWALS]
+        : [TableType.PURCHASES, TableType.WITHDRAWALS]
+      setTableOptions(tableOptions)
+      setSelectedTable(tableView ?? tableOptions[0])
+    }
+  }, [accountHasTracks, setSelectedTable, tableView, setTableOptions])
```

Comment:
nit: should we pull this out to the root component instead of duplicating code? more plumbing but i think cleaner?